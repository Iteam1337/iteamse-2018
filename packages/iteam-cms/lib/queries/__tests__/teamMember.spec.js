const resolver = require('../teamMember')

describe('queries/teamMember', () => {
  let context
  let variables

  beforeEach(() => {
    variables = {
      shortName: 'rln',
    }

    context = {
      getByContentType: jest.fn().mockResolvedValue([
        {
          sys: {
            space: [{}],
            id: '2eap07OhAYIaU00kEMaSqs',
            type: 'Entry',
            createdAt: '2018-03-23T10:40:35.562Z',
            updatedAt: '2018-03-24T07:37:49.839Z',
            revision: 2,
            contentType: [{}],
            locale: 'sv-SE',
          },
          fields: {
            name: 'Rickard Laurin',
            title: 'Webbutvecklare',
            short: 'rln',
            location: 'Stockholm',
            email: 'rickard.laurin@iteam.se',
          },
        },
      ]),
    }
  })

  it('should call getContentType with correct content type', async () => {
    try {
      await resolver(null, variables, context)

      expect(context.getByContentType).toHaveBeenCalledWith('teamMembers', {
        'fields.short': 'rln',
      })
    } catch (error) {
      throw new Error(error)
    }
  })

  it('should return all fields', async () => {
    await expect(resolver(null, variables, context)).resolves.toMatchSnapshot()
  })

  it('should handle errors', async () => {
    expect.hasAssertions()
    context.getByContentType.mockRejectedValue('err')

    await expect(resolver(null, variables, context)).rejects.toThrowError('err')
  })
})
