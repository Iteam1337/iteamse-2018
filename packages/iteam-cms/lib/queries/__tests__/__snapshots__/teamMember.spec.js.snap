// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/teamMember should return all fields 1`] = `
Object {
  "avatar": "https://www.gravatar.com/avatar/08053b9b0035a1ad64dc9be5485f2cf5?s=300",
  "email": "rickard.laurin@iteam.se",
  "headerImage": undefined,
  "location": "Stockholm",
  "name": "Rickard Laurin",
  "short": "rln",
  "title": "Webbutvecklare",
}
`;
